toolchain/check/import_ref.cpp

```diff
@@ -1069,6 +1056,28 @@ static auto GetLocalNameScopeId(ImportRefResolver& resolver,
                name_scope_inst);
 }
 
+// Translates a NameScopeId from the import IR to a local NameScopeId. Adds
+// unresolved constants to the resolver's work stack.
+static auto GetLocalNameScopeId(ImportRefResolver& resolver,
+                                SemIR::NameScopeId name_scope_id)
+    -> SemIR::NameScopeId {
+  // Get the instruction that created the scope.
+  auto [inst_id, inst] =
+      resolver.import_name_scopes().GetInstIfValid(name_scope_id);
+  if (!inst) {
+    // Map scopes that aren't associated with an instruction to `None`. For now,
+    // such scopes aren't used, and we don't have a good way to remap them.
+    return SemIR::NameScopeId::None;
+  }
+
+  // Get the constant value for the scope.
+  auto const_id = GetLocalConstantId(resolver, inst_id);
+  auto result = GetLocalNameScopeIdImpl(resolver, const_id);
+  CARBON_CHECK((name_scope_id.has_value() == result.has_value()) ||
+               resolver.HasNewWork());
```

Comment:
We know that `name_scope_id.has_value()` here, because if not, `GetInstIfValid` would have returned `{None, nullopt}`, so I think we can simplify this to just
```suggestion
  CARBON_CHECK(result.has_value() || resolver.HasNewWork());
```
But see my other comment :)